File: Controlador/UsuarioControlador.py
class UsuarioControlador:
    def __init__(self, app):
        self.app = app
        self.panel = None

    def setPanel(self, panel):
        self.panel = panel

    def guardarUsuario(self, u):
        existe = any(x.getId() == u.getId() for x in self.app.getUsuarios())
        if existe:
            self.app.getUsuarioBD().actualizar(u)
        else:
            self.app.getUsuarioBD().insertar(u)
        
        if self.panel is not None:
            self.panel.refrescar()

    def eliminarUsuario(self, id_usr):
        if self.app.getUsuarioActivo() and id_usr == self.app.getUsuarioActivo().getId():
            return
        self.app.getUsuarioBD().eliminar(id_usr)
        if self.panel is not None:
            self.panel.refrescar()

    def generarId(self):
        return 0

    def getUsuarios(self):
        return self.app.getUsuarios()

    def getUsuarioActivo(self):
        return self.app.getUsuarioActivo()

File: Controlador/UsuarioControlador.cs
using System.Collections.Generic;
using System.Linq;
using punto_de_venta_C_.Modelo;
using punto_de_venta_C_.Vista;

namespace punto_de_venta_C_.Controlador
{
    public class UsuarioControlador
    {
        private readonly ControladorPrincipal app;
        private UsuariosPanel panel;

        public UsuarioControlador(ControladorPrincipal app)
        {
            this.app = app;
        }

        public void setPanel(UsuariosPanel panel)
        {
            this.panel = panel;
        }

        public void guardarUsuario(Usuario u)
        {
            bool existe = app.getUsuarios().Any(x => x.getId() == u.getId());
            if (existe)
            {
                app.getUsuarioBD().actualizar(u);
            }
            else
            {
                app.getUsuarioBD().insertar(u);
            }
            if (panel != null)
                panel.refrescar();
        }

        public void eliminarUsuario(int id)
        {
            if (id == app.getUsuarioActivo().getId())
                return;
            app.getUsuarioBD().eliminar(id);
            if (panel != null)
                panel.refrescar();
        }

        public int generarId()
        {
            return 0;
        }

        public List<Usuario> getUsuarios()
        {
            return app.getUsuarios();
        }

        public Usuario getUsuarioActivo()
        {
            return app.getUsuarioActivo();
        }
    }
}
